presto-main/src/main/java/com/facebook/presto/operator/aggregation/MapUnionSumAggregation.java

```diff
@@ -0,0 +1,151 @@
+
+    public MapUnionSumAggregation()
+    {
+        super(NAME,
+                ImmutableList.of(comparableTypeParameter("K"), nonDecimalNumericTypeParameter("V")),
+                ImmutableList.of(),
+                parseTypeSignature("map<K,V>"),
+                ImmutableList.of(parseTypeSignature("map<K,V>")));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Aggregate all the maps into a single map summing the values for matching keys";
+    }
+
+    @Override
+    public InternalAggregationFunction specialize(BoundVariables boundVariables, int arity, FunctionAndTypeManager functionAndTypeManager)
+    {
+        Type keyType = boundVariables.getTypeVariable("K");
+        Type valueType = boundVariables.getTypeVariable("V");
+        MapType outputType = (MapType) functionAndTypeManager.getParameterizedType(MAP, ImmutableList.of(
+                TypeSignatureParameter.of(keyType.getTypeSignature()),
+                TypeSignatureParameter.of(valueType.getTypeSignature())));
+
+        return generateAggregation(keyType, valueType, outputType);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type keyType, Type valueType, MapType outputType)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(MapUnionSumAggregation.class.getClassLoader());
+        List<Type> inputTypes = ImmutableList.of(outputType);
+        MapUnionSumStateSerializer stateSerializer = new MapUnionSumStateSerializer(outputType);
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, outputType.getTypeSignature(), inputTypes.stream().map(Type::getTypeSignature).collect(toImmutableList())),
+                createInputParameterMetadata(outputType),
+                INPUT_FUNCTION.bindTo(keyType).bindTo(valueType),
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                ImmutableList.of(new AccumulatorStateDescriptor(
+                        MapUnionSumState.class,
+                        stateSerializer,
+                        new MapUnionSumStateFactory(keyType, valueType))),
+                outputType);
+
+        GenericAccumulatorFactoryBinder factory = AccumulatorCompiler.generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, ImmutableList.of(intermediateType), outputType, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type inputType)
+    {
+        return ImmutableList.of(
+                new ParameterMetadata(STATE),
+                new ParameterMetadata(INPUT_CHANNEL, inputType));
+    }
+
+    public static void input(Type keyType, Type valueType, MapUnionSumState state, Block value)
```

Comment:
Can we change the `Block value` to `Block input`? value and valueBlock/valueType is totally different which makes it very confusing.

Reply:
Done

Reply:
not done...

Reply:
Done for real this time :)